script.js

```diff
@@ -0,0 +1,218 @@
+      { text: '1 + 2 + "3" //output "123"', isCorrect: false },
+      { text: "[,*,].length", isCorrect: false },
+      { text: ";let a = 2", isCorrect: true },
+    ],
+  },
+  {
+    question: "How many years in the row JavaScript keep being the most popular programming language between developers?",
+    options: [
+      { text: "10 years", isCorrect: false },
+      { text: "11 years", isCorrect: true },
+      { text: "12 years", isCorrect: false },
+      { text: "13 years", isCorrect: false },
+    ],
+  },
+];
+
+function filterQuestions() {
+  const contentInput = document.getElementById("content-input").value;
+  const filteredQuestions = quizData.filter((question) =>
+    question.question.toLowerCase().includes(contentInput.toLowerCase())
+  );
+
+  const filteredQuestionsDiv = document.querySelector(".filteredQuestions");
+  filteredQuestionsDiv.innerHTML = "";
+
+  if (filteredQuestions.length === 0) {
+    const noResultMessage = document.createElement("p");
+    noResultMessage.textContent = "No questions found";
+    filteredQuestionsDiv.appendChild(noResultMessage);
+  } else {
+    filteredQuestions.forEach((question) => {
+      const questionElement = document.createElement("div");
+      questionElement.classList.add("question");
+      
+      const questionText = document.createElement("p");
+      questionText.textContent = question.question;
+      questionText.style.fontWeight = "bold";
+      questionElement.appendChild(questionText);
+      questionElement.style.marginTop = "10px";
+
+      const optionsElement = document.createElement("ol");
+      optionsElement.classList.add("options");
+      optionsElement.style.padding = "10px";
+
+      question.options.forEach((option) => {
+        const optionElement = document.createElement("li");
+        optionElement.textContent = option.text;
+        optionsElement.appendChild(optionElement);
+      });
+
+      questionElement.appendChild(optionsElement);
+      filteredQuestionsDiv.appendChild(questionElement);
+    });
+  }
+}
+
+function startQuiz() {
+  const player1 = document.getElementById('name-input1').value;
+  const player2 = document.getElementById('name-input2').value;
+  
+  const playersList = document.getElementById('playersList');
+  playersList.innerHTML = 
+  '<div>' + player1 + '<span id="scorePlayer1">0</span><button type="button" onclick="addScore(1, true)">Correct</button><button type="button" onclick="addScore(1, false)">Wrong</button></div>' +
+  '<div>' + player2 + '<span id="scorePlayer2">0</span><button type="button" onclick="addScore(2, true)">Correct</button><button type="button" onclick="addScore(2, false)">Wrong</button></div>';
+}
+
+function addScore(playerNumber, isCorrect) {
+  let scoreElement = document.getElementById('scorePlayer' + playerNumber);
+  let score = parseInt(scoreElement.innerHTML);
+  score += isCorrect ? 1 : -1;
+  scoreElement.innerHTML = score;
+}
+
+async function getData() {
```

Comment:
Instead of awaiting each step, you can use this syntax:

`fetch('YOUR_LINK')
  .then((response) => response.json())
  .then((data) => {
    //handle your data here
  })
  .catch((error) => {
    //handle potential errors here
  });`
  
You can then add this data to your questionsArray, so that the data is already populated from the API when the page is loaded, which will happen if you place the fetch outside the function, it will happen without you having to call the getData function 👍